core/src/main/scala/org/apache/spark/rpc/RpcEndpointRef.scala

```diff
@@ -92,4 +92,18 @@ private[spark] abstract class RpcEndpointRef(conf: SparkConf)
     timeout.awaitResult(future)
   }
 
+  def askSyncWithInterruptCheck[T: ClassTag](
+      message: Any,
+      timeout: RpcTimeout,
+      interruptCheck: () => Unit): T = {
+    val future = ask[T](message, timeout)
+    while(!future.isCompleted) {
+      interruptCheck()
+      Thread.sleep(10)
```

Comment:
Um, so there still has the possibility(very unlikely) if the tasks are killed during the 10ms?